app/models/reset_code.rb

```diff
@@ -0,0 +1,23 @@
+class ResetCode < ActiveRecord::Base
+
+  DEFAULT_EXPIRATION_PERIOD = \
+    Rails.configuration.accounts.default_reset_code_expiration_period
+
+  belongs_to :identity, inverse_of: :reset_code
+
+  validates :identity, presence: true
+  validates :identity_id, uniqueness: true
+  validates :code, presence: true, uniqueness: true
+
+  def generate(expiration_period = DEFAULT_EXPIRATION_PERIOD)
+    self.code = SecureRandom.hex(16)
+    self.expires_at = expiration_period.nil? ? nil : \
+                                               DateTime.now + expiration_period
+  end
+
+  # Returns true if the code is present in the DB and hasn't expired
```

Comment:
this comment talks about the `code` but the method doesn't do anything with `code`